src/cascadia/TerminalApp/CommandPalette.cpp

```diff
@@ -784,9 +784,13 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
-    void CommandPalette::EnableCommandPaletteMode()
+    void CommandPalette::EnableCommandPaletteMode(CommandPaletteLaunchMode const& launchMode)
     {
-        _switchToMode(CommandPaletteMode::ActionMode);
+        auto mode = (launchMode == CommandPaletteLaunchMode::CommandLine) ?
+                        CommandPaletteMode::CommandlineMode :
+                        CommandPaletteMode::ActionMode;
```

Comment:
Could/Should we consolidate these into one `CommandPaletteMode` enum in ActionArgs.idl? That way we don't have to do this translation? Or is that a weird thing to do since the mode that the command palette is in isn't really a setting. (this one is more of a thought exercise than a "requested change")

Reply:
Thought exercise already completed in https://github.com/microsoft/terminal/issues/8322#issuecomment-729976112 and https://github.com/microsoft/terminal/issues/8322#issuecomment-729979167

Reply:
As Dustin mentioned, we didn't want to expose the entire enum, so I thought that the introduction of the new one was the simplest way to achieve this.